cli/src/pcluster/aws/aws_resources.py

```diff
@@ -230,11 +230,7 @@ def is_cpu_options_supported_in_lt(self):
                 # If default threads per core is 1, HT doesn't need to be disabled
                 self.default_threads_per_core() > 1,
                 # Currently, hyperthreading must be disabled manually on *.metal instances
-                not (
-                    instance_type.endswith(".metal")
-                    or instance_type.startswith("m4.")
-                    or instance_type in ["cc2.8xlarge"]
-                ),
+                not (instance_type.endswith(".metal") or instance_type in ["cc2.8xlarge"]),
```

Comment:
I think before removing the hardcoding we should have changed the way we decide whether disabling HT through CPU options or not.
In particular we need to update the code here so that we check that DefaultThreadsPerCore > 1 and that ValidThreadsPerCore contains 1. when CpuOptions is not supported you won’t find ValidThreadsPerCore but you’ll still have DefaultThreadsPerCore